control-plane-operator/hostedclusterconfigoperator/controllers/machine/machine.go

```diff
@@ -0,0 +1,137 @@
+	discoveryv1 "k8s.io/api/discovery/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+	capiv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+func (r *reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reconcile.Result, error) {
+	log := ctrl.LoggerFrom(ctx)
+	log.Info("Reconciling")
+	machine := &capiv1.Machine{}
+	if err := r.client.Get(ctx, client.ObjectKey(req.NamespacedName), machine); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("not found", "Machine", r)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get Machine: %w", err)
+	}
+
+	hcp := &hyperv1.HostedControlPlane{}
+	if err := r.client.Get(ctx, r.hcpKey, hcp); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("not found", "HostedControlPlane", r.hcpKey)
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, fmt.Errorf("failed to get HostedControlPlane: %w", err)
+	}
+	switch hcp.Spec.Platform.Type {
+	case hyperv1.KubevirtPlatform:
+		if hcp.Spec.Platform.Kubevirt != nil && (hcp.Spec.Platform.Kubevirt.BaseDomainPassthrough != nil && *hcp.Spec.Platform.Kubevirt.BaseDomainPassthrough) {
+			if err := r.reconcileKubevirtDefaultIngressEndpoints(ctx, hcp, machine); err != nil {
+				return reconcile.Result{}, fmt.Errorf("failed reconciling kubevirt default ingress passthrough service endpoints: %w", err)
+			}
+		}
+	}
+	log.Info("Reconciled Machine")
+	return reconcile.Result{}, nil
+}
+
+func (r *reconciler) reconcileKubevirtDefaultIngressEndpoints(ctx context.Context, hcp *hyperv1.HostedControlPlane, machine *capiv1.Machine) error {
+	log := ctrl.LoggerFrom(ctx)
+	var namespace string
+	if hcp.Spec.Platform.Kubevirt.Credentials != nil {
+		namespace = hcp.Spec.Platform.Kubevirt.Credentials.InfraNamespace
+	} else {
+		namespace = hcp.Namespace
+	}
+
+	// Manifests for infra/mgmt cluster passthrough service
+	cpService := hcpmanifests.IngressDefaultIngressPassthroughService(namespace)
+
+	cpService.Name = fmt.Sprintf("%s-%s",
+		hcpmanifests.IngressDefaultIngressPassthroughServiceName,
+		hcp.Spec.Platform.Kubevirt.GenerateID)
+
+	if err := r.kubevirtInfraClient.Get(ctx, client.ObjectKeyFromObject(cpService), cpService); err != nil {
+		return fmt.Errorf("failed to get default ingress passthrow Service: %w", err)
+	}
+
+	// If there is a selector endpoints should not be generated
+	if len(cpService.Spec.Selector) > 0 {
+		return nil
+	}
+
+	if len(cpService.Spec.Ports) == 0 {
+		return fmt.Errorf("missing default ingress passthrow Service %s/%s ports", cpService.Namespace, cpService.Name)
+	}
+
+	ipv4EndpointSlice := &discoveryv1.EndpointSlice{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: cpService.Namespace,
+			Name:      cpService.Name + "-" + machine.Name + "-ipv4",
+			Labels: map[string]string{
```

Comment:
I don't think this is strictly necessary, but according to the documentation [here](https://kubernetes.io/docs/concepts/services-networking/endpoint-slices/#management) we should be setting the  `endpointslice.kubernetes.io/managed-by` label

Reply:
the service-name label is mandatory to map endpointslices to the corresponding service, I will add the managed-by label though.